app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -152,7 +152,8 @@ function insertArticles(articles) {
   if (el) {el.outerHTML = "";}
   articles.forEach(function insertAnArticle(article) {
     var existingEl = document.getElementById("article-link-"+article.id);
-    if ( existingEl &&
+    if ( !['/', '/top/week', '/top/month', '/top/year', '/top/infinity', '/latest'].includes(window.location.pathname) &&
```

Comment:
This prevents articles from being overwritten with potentially newer Algolia data. As discussed @benhalpern, in the case of the home page, this is not necessary since the new feed endpoint has up to feed data already.